### PR TITLE
Feature/se 57

### LINKED ISSUE #57: Create the front-end for Job profile describe the jobs
Job profile describe the jobs

### PR DESCRIPTION
**Pull request by 11Meru11**
close SE2-WS23/documentation#57

page to select the neccessary skills for the job

### CODE REVIEW

File: src/pages/JobDescribtion/components/JobDescriptionItem.jsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  ToggleButton,
+  Grid,
+  ToggleButtonGroup,
+  ListItem,
+  ListItemText,
+} from "@mui/material";
+import PropTypes from "prop-types";
+import React from "react";
+
+function JobDescriptionItem(props) {
```
Comment (Chrissydrx) [inline src/pages/JobDescribtion/components/JobDescriptionItem.jsx]: Please add JSDoc

File: src/pages/JobDescribtion/JobDescriptionPage.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          display: "flex",
+          width: "175px",
+          height: "43",
+          padding: "6px 16px",
+          justifyContent: "center",
+          alignItems: "center",
+          left: "75%",
+          marginTop: "20%",
+          marginLeft: "60px",
+          marginBottom: "50px",
```
Comment (Chrissydrx) [inline src/pages/JobDescribtion/JobDescriptionPage.jsx]: Why did you use so much styling for this button? Other people have done something similar in their PR and I think so much styling is not necessary

File: src/pages/JobDescribtion/JobDescriptionPage.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+          height: "59px",
+          flexShrink: "0",
+          fontSize: "30px",
+          fontFamily: "Inter",
+          fontStyle: "normal",
+          fontWeight: "400",
+          lineHeight: "normal",
+          marginTop: "100px",
+          marginLeft: "auto",
+          marginRight: "auto",
```
Comment (Chrissydrx) [inline src/pages/JobDescribtion/JobDescriptionPage.jsx]: I think most of it should be already in the Theme. With this approach you override that and it is hard to maintain changes in the theme with this approach

File: src/pages/JobDescribtion/JobDescriptionPage.jsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
+import { Button, Grid, Typography } from "@mui/material/";
+import { v4 as uuidv4 } from "uuid";
+import PageLayout from "../../components/PageLayout";
+import JobDescriptionItem from "./components/JobDescriptionItem";
+import mockedDescription from "./mockData";
+
+function JobDescriptionPage() {
```
Comment (Chrissydrx) [inline src/pages/JobDescribtion/JobDescriptionPage.jsx]: Please add JSDoc

File: src/main.jsx [hunk 5]
```diff
@@ -27,6 +27,7 @@ const router = createBrowserRouter([
     path: "/",
     errorElement: <ErrorPage />,
     children: ROUTES,
+
```
Comment (Chrissydrx) [inline src/main.jsx]: Why did you commit this?

File: src/pages/JobDescribtion/JobDescriptionPage.jsx
Comment (Chrissydrx) [inline src/pages/JobDescribtion/JobDescriptionPage.jsx:1]: Spelling mistake in the folder name

File: src/pages/JobDescribtion/components/JobDescriptionItem.jsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  const [selected, setSelected] = React.useState(() => [0]);
+  const handleChange = (_, newSelected) => {
+    setSelected(newSelected);
+  };
```
Comment (Chrissydrx) [inline src/pages/JobDescribtion/components/JobDescriptionItem.jsx]: I think with this approach you can not use that state in the Page itself. You should pass the state to this component from the Page.

File: src/pages/JobDescribtion/components/JobDescriptionItem.jsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  Grid,
+  ToggleButtonGroup,
+  ListItem,
+  ListItemText,
+} from "@mui/material";
+import PropTypes from "prop-types";
+import React from "react";
+
+function JobDescriptionItem(props) {
+  const [selected, setSelected] = React.useState(() => [0]);
```
Comment (Chrissydrx) [inline src/pages/JobDescribtion/components/JobDescriptionItem.jsx]: What kind of default is this?
Is it not the same as [0]? And why is this an array?

File: src/pages/JobDescribtion/components/JobDescriptionItem.jsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+      <ToggleButtonGroup value={selected} onChange={handleChange}>
+        <ToggleButton
+          value="true"
+          sx={{
+            height: "50px",
+            width: "150px",
+            background: useTheme().palette.primary.light,
+          }}
+        >
+          <ListItem sx={{ textAlign: "center", fontSize: "60px" }}>
```
Comment (Chrissydrx) [inline src/pages/JobDescribtion/components/JobDescriptionItem.jsx]: Use theme for fontSize please

File: src/pages/JobDescribtion/mockData.js
Comment (Chrissydrx) [inline src/pages/JobDescribtion/mockData.js:1]: Is it not possible to just have an array? I mean your object has only one key

File: src/pages/JobDescribtion/components/JobDescriptionItem.jsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+          </ListItem>
+        </ToggleButton>
+      </ToggleButtonGroup>
+    </Grid>
+  );
+}
+
+JobDescriptionItem.propTypes = {
+  jobDescription: PropTypes.shape({
+    skills: PropTypes.string,
```
Comment (Chrissydrx) [inline src/pages/JobDescribtion/components/JobDescriptionItem.jsx]: Why is this key called skills? If I'm not mistaken, skills is only a string and represents one skill, so skill would be better, if you don't want to replace that skill object (see comment mock data)

**Review by Chrissydrx — CHANGES_REQUESTED** [review]

File: src/main.jsx (hunk 5 above)
Comment (11Meru11) [reply src/main.jsx]: when i tried to merge it showed me there merge problems

File: src/pages/JobDescribtion/JobDescriptionPage.jsx
Comment (11Meru11) [reply src/pages/JobDescribtion/JobDescriptionPage.jsx:1]: cant change the name. showed it multiple times, then its loading then it throws the error Error: EPERM: operation not permitted, rename 'c:\Users\Nicolas\SE3 project code (important)\Skillverse-Web\src\pages\JobDescribtion' -> 'c:\Users\Nicolas\SE3 project code (important)\Skillverse-Web\src\pages\JobDescription'

File: src/pages/JobDescribtion/JobDescriptionPage.jsx (hunk 3 above)
Comment (11Meru11) [reply src/pages/JobDescribtion/JobDescriptionPage.jsx]: i used the css from Figma

File: src/pages/JobDescribtion/JobDescriptionPage.jsx (hunk 2 above)
Comment (11Meru11) [reply src/pages/JobDescribtion/JobDescriptionPage.jsx]: i used the css from figma. im gonna change it like on the other where i used to much styling

File: src/pages/JobDescribtion/components/JobDescriptionItem.jsx (hunk 7 above)
Comment (11Meru11) [reply src/pages/JobDescribtion/components/JobDescriptionItem.jsx]: it is from the mui code example side. the default is that none of the toggle is activated when the site is loaded

File: src/pages/JobDescribtion/JobDescriptionPage.jsx
Comment (Chrissydrx) [reply src/pages/JobDescribtion/JobDescriptionPage.jsx:1]: I fixed that for you

File: src/pages/JobDescribtion/components/JobDescriptionItem.jsx (hunk 7 above)
Comment (Chrissydrx) [reply src/pages/JobDescribtion/components/JobDescriptionItem.jsx]: In the MUI Documentation is this the default value
image

File: src/main.jsx (hunk 5 above)
Comment (Chrissydrx) [reply src/main.jsx]: That is an empty line

**Comment by Chrissydrx** [comment]
**@11Meru11 Your commit messages are not in our schema. Please stick to our format Verb: + Content and please write them in English. Next time we will close that PR.**

File: src/pages/JobDescription/JobDescriptionPage.jsx [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+import { v4 as uuidv4 } from "uuid";
+import PageLayout from "../../components/PageLayout";
+import JobDescriptionItem from "./components/JobDescriptionItem";
+import mockedSkills from "./mockData";
+import { useTheme } from "@emotion/react";
+/**
+ * A component which renders the JobDescriptionPage to let the company select the necessary skills for the job
+ * @returns {JSX.Element} The JobDescriptionPage component.
+ */
+
```
Comment (julix14) [inline src/pages/JobDescription/JobDescriptionPage.jsx:11]:
```suggestion
```

File: src/pages/JobDescription/JobDescriptionPage.jsx [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+      >
+        Select relevant Skills describing the job
+      </Typography>
+      <Grid
+        container
+        rowSpacing={15}
+        columnSpacing={10}
+        columns={16}
+        sx={{
+          display: "flex",
```
Comment (julix14) [inline src/pages/JobDescription/JobDescriptionPage.jsx:30]: You mix grid and flex here, stick to one of both

File: src/pages/JobDescription/JobDescriptionPage.jsx [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+          justifyContent: "center",
+          marginTop: "50px",
+          marginBottom: "50px",
+        }}
+      >
+        {mockedSkills?.map((JobDescription) => (
+          <JobDescriptionItem mockedSkills={JobDescription} key={uuidv4()} />
```
Comment (julix14) [inline src/pages/JobDescription/JobDescriptionPage.jsx:37]:
```suggestion
        {mockedSkills?.map((jobDescription) => (
          <JobDescriptionItem mockedSkills={jobDescription} key={uuidv4()} />
```

Stick to the common variable naming conventions

File: src/pages/JobDescription/components/JobDescriptionItem.jsx [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+import React from "react";
+import PropTypes from "prop-types";
+/**
+ * A component that displays a training block with course name, skills, progress, and description.
+ * @param {Object} props - The props object.
+ * @param {Object} props.JobDescription - The training object containing course name, skills, progress, description, and status.
+ * @param {string} props.mockedSkills - Prints the Skills in the toggles
+ * @returns {JSX.Element} the JobDescriptionPage Component.
+ */
+
```
Comment (julix14) [inline src/pages/JobDescription/components/JobDescriptionItem.jsx:12]:
```suggestion
```

File: src/pages/JobDescription/components/JobDescriptionItem.jsx [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+          <ListItemText>{props.mockedSkills}</ListItemText>
+        </ListItem>
+      </ToggleButton>
+    </Grid>
+  );
+}
+
+JobDescriptionItem.propTypes = {
+  mockedSkills: PropTypes.string,
+};
```
Comment (julix14) [inline src/pages/JobDescription/components/JobDescriptionItem.jsx:39]: It is only one skill that is passed, so the naming doesn't fit here.
And it also shouldn't be called "mocked..."

File: src/pages/JobDescription/JobDescriptionPage.jsx [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+      </Typography>
+      <Grid
+        container
+        rowSpacing={15}
+        columnSpacing={10}
+        columns={16}
+        sx={{
+          display: "flex",
+          justifyContent: "center",
+          marginTop: "50px",
```
Comment (julix14) [inline src/pages/JobDescription/JobDescriptionPage.jsx:32]: Why so much spacing here?

**Comment by Chrissydrx** [comment]
image
Items are not centered

File: src/pages/JobDescription/JobDescriptionPage.jsx [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+ * @returns {JSX.Element} The JobDescriptionPage component.
+ */
+
+function JobDescriptionPage() {
+  return (
+    <PageLayout title="Job Description">
+      <Typography
+        variant="h1"
+        sx={{
+          margin: "100px 100px",
```
Comment (julix14) [inline src/pages/JobDescription/JobDescriptionPage.jsx:18]: Why so much spacing here?

File: src/pages/JobDescription/JobDescriptionPage.jsx [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+/**
+ * A component which renders the JobDescriptionPage to let the company select the necessary skills for the job
+ * @returns {JSX.Element} The JobDescriptionPage component.
+ */
+
+function JobDescriptionPage() {
+  return (
+    <PageLayout title="Job Description">
+      <Typography
+        variant="h1"
```
Comment (julix14) [inline src/pages/JobDescription/JobDescriptionPage.jsx:16]: Not the required font size

File: src/pages/JobDescription/JobDescriptionPage.jsx [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+        sx={{
+          margin: "100px 100px",
+          fontSize: useTheme().typography,
```
Comment (julix14) [inline src/pages/JobDescription/JobDescriptionPage.jsx:19]: Wrong use of theme here, and it is overwriting the variant

**Comment by Chrissydrx** [comment]
image
Text in Buttons is not gecentered

File: src/routes.jsx [hunk 19]
```diff
@@ -5,11 +5,11 @@ import CreateJobProfilePage from "./pages/CreateJobProfile/CreateJobProfilePage"
 import DashboardPage from "./pages/Dashboard/DashboardPage";
 import EmployeeOverviewPage from "./pages/EmployeeOverview/EmployeeOverviewPage";
 import HomePage from "./pages/Home/HomePage";
+import JobDescriptionPage from "./pages/JobDescription/JobDescriptionPage";
 import JobProfilePage from "./pages/JobProfile/JobProfilePage";
 import LinkedInLearning from "./pages/LinkedInLearning/LinkedInLearningPage";
 import LoginPage from "./pages/Login/LoginPage";
 import TrainingPlanPage from "./pages/TrainingPlan/TrainingPlanPage";
-
```
Comment (julix14) [inline src/routes.jsx:12]: Why did you deleted this line?

File: src/pages/JobDescription/components/JobDescriptionItem.jsx [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+function JobDescriptionItem(props) {
+  const [selected, setSelected] = React.useState(false);
+  return (
+    <Grid item xs={3}>
+      <ToggleButton
+        value="check"
+        selected={selected}
+        onChange={() => {
+          setSelected(!selected);
+        }}
```
Comment (julix14) [inline src/pages/JobDescription/components/JobDescriptionItem.jsx:22]: Use handle function for this

**Review by julix14 — CHANGES_REQUESTED** [review]
It it not working as expected, because the state will not passed from the Item to the page.
There many (layout) issues and the commits are not named as we agreed on

**Comment by julix14** [comment]
We have 35 conversations on this PR and there is no end in sight.
So we close this PR at this point, because it has >30 conversations open
Fix all the issues and adapt the page to the mockup.

**Pull request closed without merging**